frame/nfts/src/lib.rs

```diff
@@ -974,7 +1003,66 @@ pub mod pallet {
 			origin: OriginFor<T>,
 			collection: T::CollectionId,
 			item: T::ItemId,
-			maybe_check_delegate: Option<AccountIdLookupOf<T>>,
+			delegate: AccountIdLookupOf<T>,
+		) -> DispatchResult {
+			let maybe_check: Option<T::AccountId> = T::ForceOrigin::try_origin(origin)
+				.map(|_| None)
+				.or_else(|origin| ensure_signed(origin).map(Some).map_err(DispatchError::from))?;
+
+			let delegate = T::Lookup::lookup(delegate)?;
+
+			let collection_details =
+				Collection::<T, I>::get(&collection).ok_or(Error::<T, I>::UnknownCollection)?;
+			let mut details =
+				Item::<T, I>::get(&collection, &item).ok_or(Error::<T, I>::UnknownCollection)?;
+
+			let maybe_deadline =
+				details.approvals.get(&delegate).ok_or(Error::<T, I>::NotDelegate)?;
+			let is_past_deadline = if let Some(deadline) = maybe_deadline {
+				let now = frame_system::Pallet::<T>::block_number();
+				now > *deadline
+			} else {
+				false
+			};
+
+			if !is_past_deadline {
```

Comment:
Maybe we could do something like this?

Reply:
@gilescope As you mentioned it would probably be a good idea to put this into a function.

Reply:
yep, looks good to me. Pls add a new-line before `let is_past_deadline` :)